terraform/modules/tf_alert_processor_iam/main.tf

```diff
@@ -0,0 +1,105 @@
+// Permissions specific to the alert processor: decrypting secrets, sending alerts to outputs
+
+locals {
+  firehose_arn_prefix = "arn:aws:firehose:${var.region}:${var.account_id}"
+  lambda_arn_prefix   = "arn:aws:lambda:${var.region}:${var.account_id}:function"
+}
+
+resource "aws_iam_role_policy" "base" {
+  name   = "DecryptSecretsAndUseDefaultOutputs"
+  role   = "${var.role_id}"
+  policy = "${data.aws_iam_policy_document.base.json}"
+}
+
+data "aws_iam_policy_document" "base" {
+  // Allow sending alerts to default firehose output
+  statement {
+    effect    = "Allow"
+    actions   = ["firehose:Put*"]
+    resources = ["${local.firehose_arn_prefix}:deliverystream/${var.prefix}_streamalert_alert_delivery"]
+  }
+
+  // Allow decrypting output secrets
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "kms:Decrypt",
+      "kms:DescribeKey",
+    ]
+
+    resources = ["${var.kms_key_arn}"]
+  }
+
+  // Allow retrieving encrypted output secrets
+  statement {
+    effect    = "Allow"
+    actions   = ["s3:GetObject"]
+    resources = ["arn:aws:s3:::${var.prefix}.streamalert.secrets/*"]
+  }
+
+  // Allow saving alerts to the default .streamalerts bucket
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:ListBucket",
+      "s3:PutObject",
+      "s3:PutObjectAcl",
+    ]
+
+    resources = [
+      "arn:aws:s3:::${var.prefix}.streamalerts",
+      "arn:aws:s3:::${var.prefix}.streamalerts/*",
+    ]
+  }
+}
+
+// Allow the Alert Processor to invoke the configured output Lambda functions
+resource "aws_iam_role_policy" "invoke_lambda_outputs" {
+  count = "${length(var.output_lambda_functions)}"
+  name  = "LambdaInvoke_${element(var.output_lambda_functions, count.index)}"
+  role  = "${var.role_id}"
+
+  policy = <<EOF
```

Comment:
No, I tried :(

You can create multiple `data` resources with `count` as you can for other resources, but then you can't access them using the same array/splat syntax. This may be fixed in TF 0.11, I'm not sure

Reply:
Initially, I didn't think that was possible - I tried. But I found a different syntax which seems to do the trick